Yoast/wordpress-seo
# Add Slovak stemmer

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds external stemmer for Slovak.
* [Yoast SEO Premium] Introduces word form support for Slovak.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Run `yarn test` and make sure that everything passes.

**In Content-analysis app:**
**Note**: currently the content-app doesn't work to check word forms for Slovak. This step can be skipped for now.
* Set your website to Slovak (sk)
* Add a text of min 300 words in the text field
* Turn the `Use morphology` on
* Set `mačka` as the focus keyphrase
* Add these other forms of the word in the text: `mačky, mačke, mačku, mačkou, mačkám, mačkách, mačkami, mačička, mačičkou, mačičky, mačičke", mačičku, mačičkám, mačičkách, mačičkami` and make sure that all these forms are recognised.


**In Docker:**
* Build the free plugin
* Enable the `SLOVAK_SUPPORT` feature by adding `define( 'YOAST_SEO_SLOVAK_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Slovak
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `sk`):
<img width="715" alt="Screenshot 2021-06-17 at 17 04 31" src="https://user-images.githubusercontent.com/48715883/122423595-4e3f2980-cf8e-11eb-9801-ac4e5bea208a.png">

* Run this command `composer require yoast/wordpress-seo:dev-LINGO-869-add-slovak-stemmer@dev` in `wordpress-seo-premium`
* Build the premium plugin
* Reload your page
* In your console, confirm that the morphology request returns with `404`:
<img width="680" alt="Screenshot 2021-06-17 at 17 01 27" src="https://user-images.githubusercontent.com/48715883/122423972-a1b17780-cf8e-11eb-90cb-c49076c98dfb.png">

* Disable the `SLOVAK_SUPPORT` by adding this  `define( 'YOAST_SEO_SLOVAK_SUPPORT', false );` to `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="711" alt="Screenshot 2021-06-17 at 17 11 03" src="https://user-images.githubusercontent.com/48715883/122424676-34eaad00-cf8f-11eb-80b0-a11d72d9b7de.png">

* Confirm that there is no morphology request
<img width="728" alt="Screenshot 2021-06-17 at 17 11 19" src="https://user-images.githubusercontent.com/48715883/122424791-49c74080-cf8f-11eb-8880-728b64259b8d.png">


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* **In Docker:**
* Build the free plugin
* Enable the `SLOVAK_SUPPORT` feature by adding `define( 'YOAST_SEO_SLOVAK_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Slovak
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `sk`):
<img width="715" alt="Screenshot 2021-06-17 at 17 04 31" src="https://user-images.githubusercontent.com/48715883/122423595-4e3f2980-cf8e-11eb-9801-ac4e5bea208a.png">

* Run this command `composer require yoast/wordpress-seo:dev-LINGO-869-add-slovak-stemmer@dev` in `wordpress-seo-premium`
* Build the premium plugin
* Reload your page
* In your console, confirm that the morphology request returns with `404`:
<img width="680" alt="Screenshot 2021-06-17 at 17 01 27" src="https://user-images.githubusercontent.com/48715883/122423972-a1b17780-cf8e-11eb-90cb-c49076c98dfb.png">

* Disable the `SLOVAK_SUPPORT` by adding this  `define( 'YOAST_SEO_SLOVAK_SUPPORT', false );` to `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="711" alt="Screenshot 2021-06-17 at 17 11 03" src="https://user-images.githubusercontent.com/48715883/122424676-34eaad00-cf8f-11eb-80b0-a11d72d9b7de.png">

* Confirm that there is no morphology request
<img width="728" alt="Screenshot 2021-06-17 at 17 11 19" src="https://user-images.githubusercontent.com/48715883/122424791-49c74080-cf8f-11eb-8880-728b64259b8d.png">

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-869
